core/src/core/transaction.rs

```diff
@@ -556,6 +570,121 @@ pub fn aggregate(transactions: Vec<Transaction>) -> Result<Transaction, Error> {
 	Ok(tx.with_offset(total_kernel_offset))
 }
 
+/// Aggregate a vec of transactions into a multi-kernel transaction
+pub fn aggregate(transactions: Vec<Transaction>) -> Result<Transaction, Error> {
```

Comment:
What's the situation where we want to aggregate but we _don't_ want to apply cut-through?
Can we not just always apply cut-through?

Reply:
This function is used internally in the deaggregate function. 
For example, we receive (tx1,tx2,tx3) and we have (tx1)(tx2) in our pool. It allows us to quickly aggregate the last two txs into (tx1,tx2) for faster comparison.